crates/tools/trust_db_util.rs

```diff
@@ -0,0 +1,352 @@
+    force: bool,
+
+    /// use dpkg to generate db
+    #[clap(long)]
+    dpkg: bool,
+
+    /// number of records to generate
+    #[clap(short, long)]
+    count: Option<usize>,
+
+    /// use par_iter
+    #[clap(long)]
+    par: bool,
+}
+
+#[derive(Clap)]
+struct AddRecOpts {
+    /// File to add
+    path: String,
+}
+
+#[derive(Clap)]
+struct DelRecOpts {
+    /// File to delete
+    path: String,
+}
+
+#[derive(Clap)]
+struct DumpDbOpts {
+    /// Optional file
+    #[clap(short, long)]
+    outfile: Option<String>,
+}
+
+#[derive(Clap)]
+struct SearchDbOpts {
+    /// File to delete
+    #[clap(long)]
+    key: String,
+}
+
+#[derive(Clap)]
+struct CheckDbOpts {
+    /// use par_iter
+    #[clap(long)]
+    par: bool,
+}
+
+fn main() -> Result<(), Error> {
+    let sys_conf = cfg::All::load();
+    let all_opts: Opts = Opts::parse();
+    let trust_db_path = match all_opts.dbdir {
+        Some(ref p) => Path::new(p),
+        None => Path::new(&sys_conf.system.trust_db_path),
+    };
+
+    if !trust_db_path.exists() {
+        panic!("dbdir does not exist")
+    } else if !trust_db_path.is_dir() {
+        panic!("dbdir must be a directory")
+    }
+
+    println!("opening trust db at {}", trust_db_path.to_string_lossy());
+    let env = Environment::new()
+        .set_max_dbs(1)
+        .set_map_size(104857600)
+        .open(trust_db_path)?;
+
+    match all_opts.cmd {
+        Clear(opts) => clear(opts, &sys_conf, &env),
+        Init(opts) => init(opts, &sys_conf, &env),
+        Add(opts) => add(opts, &sys_conf, &env),
+        Del(opts) => del(opts, &sys_conf, &env),
+        Dump(opts) => dump(opts, &sys_conf),
+        Search(opts) => find(opts, &sys_conf, &env),
+        Check(opts) => check(opts, &sys_conf),
+    }
+}
+
+fn clear(_: ClearOpts, _: &cfg::All, env: &Environment) -> Result<(), Error> {
+    if let Ok(db) = env.open_db(Some(TRUST_DB_NAME)) {
+        let mut tx = env.begin_rw_txn()?;
+        tx.clear_db(db)?;
+        tx.commit()?;
+    };
+
+    Ok(())
+}
+
+fn init(opts: InitOpts, cfg: &cfg::All, env: &Environment) -> Result<(), Error> {
+    if opts.force {
+        clear(ClearOpts {}, cfg, env)?;
+    }
+
+    let db = env.create_db(Some(TRUST_DB_NAME), DatabaseFlags::DUP_SORT)?;
+
+    if opts.empty {
+        return Ok(());
+    }
+
+    let t = SystemTime::now();
+    let sys = if opts.dpkg {
+        load_dpkg_trust()?
+    } else {
+        load_rpm_trust(&cfg.system.system_trust_path)?
+    };
+
+    let mut tx = env.begin_rw_txn()?;
+    for trust in &sys {
+        let v = format!("{} {} {}", 1, trust.size, trust.hash);
+        match tx.put(db, &trust.path, &v, WriteFlags::APPEND_DUP) {
+            Ok(_) => {}
+            Err(_) => println!("skipped {}", trust.path),
+        }
+    }
+    tx.commit()?;
+
+    let duration = t.elapsed().expect("timer failure");
+    println!(
+        "initialized db with {} entries in {} seconds",
+        sys.len(),
+        duration.as_secs()
+    );
```

Comment:
This is great that we're benchmarking these operations, but I think these are more appropriate to do in verbose mode instead of all the time.  If this tool were to be run as a cron job for instance, if something gets emitted, it generates emails and things.  Most utilities emit nothing and exit with a zero status code on success, unless run in verbose mode.

Reply:
Added verbose check here